tests/sanity_helpers.py

```diff
@@ -1,133 +1,73 @@
 
-    Args:
-        resources (tuple): Lists of projects, secrets, pools,
-            storageclasses, pvcs and pods
-        run_io (bool): True for run IO, False otherwise
+        """
+        logger.info(f"Creating resources and running IO as a sanity functional validation")
 
-    """
-    # Create resources and run IO for both FS and RBD
-    # Unpack resources
-    projects, secrets, pools, storageclasses, pvcs, pods = resources[:6]
-
-    # Project
-    projects.append(helpers.create_project())
-
-    # Secrets
-    secrets.append(helpers.create_secret(constants.CEPHBLOCKPOOL))
-    secrets.append(helpers.create_secret(constants.CEPHFILESYSTEM))
-
-    # Pools
-    pools.append(helpers.create_ceph_block_pool())
-    pools.append(helpers.get_cephfs_data_pool_name())
-
-    # Storageclasses
-    storageclasses.append(
-        helpers.create_storage_class(
-            interface_type=constants.CEPHBLOCKPOOL,
-            interface_name=pools[0].name,
-            secret_name=secrets[0].name
-        )
-    )
-    storageclasses.append(
-        helpers.create_storage_class(
-            interface_type=constants.CEPHFILESYSTEM,
-            interface_name=pools[1],
-            secret_name=secrets[1].name
-        )
-    )
-
-    # PVCs
-    pvcs.append(helpers.create_pvc(
-        sc_name=storageclasses[0].name, namespace=projects[0].namespace)
-    )
-    pvcs.append(helpers.create_pvc(
-        sc_name=storageclasses[1].name, namespace=projects[0].namespace)
-    )
-    for pvc in pvcs:
-        helpers.wait_for_resource_state(pvc, constants.STATUS_BOUND)
-        pvc.reload()
-
-    # Pods
-    pods.append(
-        helpers.create_pod(
-            interface_type=constants.CEPHBLOCKPOOL, pvc_name=pvcs[0].name,
-            namespace=projects[0].namespace
-        )
-    )
-    pods.append(
-        helpers.create_pod(
-            interface_type=constants.CEPHFILESYSTEM, pvc_name=pvcs[1].name,
-            namespace=projects[0].namespace
-        )
-    )
-    for pod in pods:
-        helpers.wait_for_resource_state(pod, constants.STATUS_RUNNING)
-        pod.reload()
-
-    if run_io:
-        # Run IO
-        for pod in pods:
+        for interface in [constants.CEPHBLOCKPOOL, constants.CEPHFILESYSTEM]:
+            self.pod_objs = pod_factory(interface)
+        for pod in self.pod_objs:
             pod.run_io('fs', '1G')
-        for pod in pods:
-            fio_result = pod.get_fio_results()
-            logger.info(f"IOPs after FIO for pod {pod.name}:")
-            logger.info(
-                f"Read: {fio_result.get('jobs')[0].get('read').get('iops')}"
-            )
-            logger.info(
-                f"Write: {fio_result.get('jobs')[0].get('write').get('iops')}"
-            )
 
+        if run_io:
+            for pod in self.pod_objs:
+                get_fio_rw_iops(pod)
 
-def delete_resources(resources):
-    """
-    Sanity validation - Delete resources (FS and RBD)
+    def delete_resources(self):
```

Comment:
if I understand correctly, this will delete resources created initially from pod_factory.
Isn't it what the pod_factory should take care of?

Reply:
the idea is that in some test cases, I would like to delete resources as part of the test itself (delete should also be tested for cluster functionality validation).